src/fromager/resolver.py

```diff
@@ -509,3 +509,52 @@ def _find_tags(
                 yield entry["tarball_url"], version
             # pagination links
             nexturl = resp.links.get("next", {}).get("url")
+
+
+class CustomPackageProvider(BaseProvider):
+    """Get best match"""
+
+    def __init__(
+        self,
+        req: Requirement,
+        constraints: Constraints,
+        versions: dict[Version, str],
+    ):
+        super().__init__(constraints=constraints)
+        self.req = req
+        self.versions = versions
+
+    def get_cache(self) -> dict[str, list[Candidate]]:
+        return GenericProvider.generic_resolver_cache
+
+    def find_best_match(
+        self,
+    ) -> Version:
+        # convert self.versions to candidates
+        candidates = []
+        for version, _ in self.versions.items():
+            # Candidates require a URL for initialization: Use localhost as a place holder
+            # Candidates must match the requirement name in is_satisfied_call below
+            candidate = Candidate(self.req.name, version, "https://localhost")
+            candidates.append(candidate)
+
+        constraint = self.constraints.get_constraint(self.req.name)
+        constraintspec = ""
+        if constraint is not None:
+            constraintspec = str(constraint.specifier)
+
+        for candidate in reversed(candidates):
```

Comment:
The list is reversed here, so I think that original loop does want some order applied.

Reply:
Yup see above comment.